### ISSUE TITLE
[Improvement]: add search bar to players list & reactions

### ISSUE THREAD

**Issue by lonelybud** [Yimura/YimMenu#2562]
### Which feature

players list eg:
https://github.com/lonelybud/lonelyMenu/blob/a5af2840ad40e11439db59873dbd2b05d05ba315/src/views/players/view_players.cpp#L120-L151

![image](https://github.com/YimMenu/YimMenu/assets/139460769/838e526c-8586-447b-ab0f-414b7aa00a9b)

reactions list eg:
https://github.com/lonelybud/lonelyMenu/blob/master/src/views/settings/view_reaction_settings.cpp

![image](https://github.com/YimMenu/YimMenu/assets/139460769/6a179d74-d591-44e3-b0ed-91d8674ade7e)

### Reason

improve ux

**Comment by TheGreenBandit** [comment]
players list should be easy however reactions will need to add a vector off each, with matching strings

**Comment by TheGreenBandit** [comment]
> players list should be easy however reactions will need to add a std::to_list, with matching strings

However this wouldnt work as the different langauges would make us have to do a different string or something for eahh.

**Comment by TheGreenBandit** [comment]
> However this wouldnt work as the different langauges would make us have to do a different string or something for eahh. 

Axtually we could translate each string and it should work lol im a dumbass

**Comment by lonelybud** [comment]
yes i understand the menu is multilingual so need to have search functionality accordingly.

I also have a concern.
given this - 
https://github.com/YimMenu/YimMenu/blob/8b62dadcd500dd6321953a6ed7a5eaf92cf4b529/src/core/settings.hpp#L220

Imagine you are not an english user, you will see english in console - 
https://github.com/YimMenu/YimMenu/blob/8b62dadcd500dd6321953a6ed7a5eaf92cf4b529/src/backend/reactions/reaction.cpp#L60

while you will see your language in notification -
https://github.com/YimMenu/YimMenu/blob/8b62dadcd500dd6321953a6ed7a5eaf92cf4b529/src/backend/reactions/reaction.cpp#L78-L79

Now if you have only enabled log and not notify then for the non-english user it will be difficult to search for that reaction (in english) in reactions settings and disable log etc.

**Comment by TheGreenBandit** [comment]
I didnt see there was constructors already for the reactions, those first strings will need to be translated accordingly and that what will be used for searching
